lib/paper_trail/reifier.rb

```diff
@@ -166,6 +167,32 @@ def reify_has_ones(transaction_id, model, options = {})
         end
       end
 
+      def reify_belongs_tos(transaction_id, model, options = {})
+        associations = model.class.reflect_on_all_associations(:belongs_to)
+
+        associations.each do |assoc|
+          next unless assoc.klass.paper_trail_enabled_for_model?
+          collection_key = model.send(assoc.association_foreign_key)
+
+          version = assoc.klass.paper_trail_version_class.
+            where("item_type = ?", assoc.class_name).
+            where("item_id = ?", collection_key).
+            where("created_at >= ? OR transaction_id = ?", options[:version_at], transaction_id).
+            order("id").limit(1).first
+
+          collection = if version.nil?
+                         assoc.klass.where(assoc.klass.primary_key => collection_key).first
+                       elsif version.event == "create"
+                         options[:mark_for_destruction] ? collection.mark_for_destruction : nil
```

Comment:
Nice catch. But I don't think this condition is ever met as the base model is always created before the associated model. Ex:- Widget instance is created before the wotsit instance.